### PR TITLE
Add queue button to click-on-DagRun interface.

### PR DESCRIPTION
**Pull request by Jorricks**
In our use-case for Airflow, we often add tasks after the DAG has been created and had a first couple successful runs.
This means, we often want successful DagRuns to be re-queued, such that it creates all the TaskInstances for the new tasks.
This option was currently not as easily accessible, only through the DagRun view.
Note: I choose to add only `Queue` and not `Running` as running could potentially go over the limit of allowed running DagRuns. 

Screenshot 2022-02-14 at 10 33 33

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by Jorricks** [comment]
Would still love some feedback. Have a good weekend all :v:

**Comment by eladkal** [comment]
> Note: I choose to add only Queue and not Running as running could potentially go over the limit of allowed running DagRuns.

I would prefer to call the action by it's name. Queue is not so self explanatory and also kinda misleading as you are not really setting the tasks to Queue state. What about "Mark Running" which in essence this is what the action does. It make the DAGRun as running and the scheduler will pick up the tasks.

**Comment by Jorricks** [comment]
Well.. I feel "Mark Running" doesn't really fit.  Marking the DagRun as Running directly could potentially overwrite the active DagRun limit (which I intentionally want to steer away from).
I don't fully agree on the statement that it might be misleading and people expect you to set all tasks to Queued. When I would put "Mark Running", you would also not expect all Tasks to be started right? So how would that be different for Queued?
However, I do agree that just "Queue" might not do the job. "Mark Queued" might be better but also not really sure...

**Comment by eladkal** [comment]
> However, I do agree that just "Queue" might not do the job. "Mark Queued" might be better but also not really sure...

The action you are taking is on DAGRun object. This is what the user expect. To my perspective since there is no such thing as Queue state for DagRuns it can cause confusion. Specifically because Task Queue state means that it already passed the scheduler and was eligible for running, waiting for worker slot - which is not what you want here.

Lets see what other thinks :)

**Comment by Jorricks** [comment]
There is a Queued DagRun state since Airflow 2.2.0 I believe.

File: airflow/www/views.py
Comment (bbovenzi) [inline airflow/www/views.py]: Let's update this description

**Comment by bbovenzi** [comment]
I think we need to explain the difference between Clear and Queue. They are both a re-run. Clear just reruns the existing task instances while Queue would add new task instances if the DAG's tasks have changed. Is that right?

**Comment by Jorricks** [comment]
> I think we need to explain the difference between Clear and Queue. They are both a re-run. Clear just reruns the existing task instances while Queue would add new task instances if the DAG's tasks have changed. Is that right?

That is a perfect description sir. However, the challenge is getting that in a couple words in a button xD

File: airflow/www/views.py
Comment (Jorricks) [reply airflow/www/views.py]:
```suggestion
        """Queue DagRun so tasks that haven't run yet can be started."""
```

**Comment by bbovenzi** [comment]
Maybe we have a new section called "Rerun" with two buttons "Soft rerun" and "Hard rerun". We can also include an info tooltip either next to "Rerun" or next to each button. Thoughts?

**Comment by Jorricks** [comment]
> Maybe we have a new section called "Rerun" with two buttons "Soft rerun" and "Hard rerun". We can also include an info tooltip either next to "Rerun" or next to each button. Thoughts?

I am a bit unsure about that.. I am not a UX designer so wouldn't know what the best way to do this is.

**Comment by bbovenzi** [comment]
> I am a bit unsure about that.. I am not a UX designer so wouldn't know what the best way to do this is.

No worries. I'll pull down your branch and give it a try.

**Comment by bbovenzi** [comment]
I updated the Run Modal UI:

Screen Shot 2022-03-04 at 12 35 14 PM

**Comment by tanelk** [comment]
You are changing the state to 'queued', but I think the correnct state would be 'scheduled'. The `SchedulerJob` should be only place, where tasks are queued.

**Comment by Jorricks** [comment]
> You are changing the state to 'queued', but I think the correnct state would be 'scheduled'. The `SchedulerJob` should be only place, where tasks are queued. 
> 
> scheduled - all task dependencies are met
> queued - all resource dependecies are met and the task is handed over to the executor.

Please read the comments above.. We are talking about changing the DagRun state, not the task..🙄

**Review by bbovenzi — APPROVED** [review]
I tested it out locally and it looks good.

But since I made a commit, we should get one more approval before merging.

**Review by eladkal — APPROVED** [review]
LGTM

**Pull request merged**
